### DISCUSSION TITLE
Fluid proposals (Osmotic Funding) 🚰

### DISCUSSION THREAD

**Discussion by kanaagawa** [1Hive/gardens#536] [Ideas]
## Summary

**A new proposal type to create flows of funds from Common Pool of Gardens.**

From our experience on running Conviction Voting on 1Hive for more than a year. We notice that not all funding proposals are equal. 

There are funding proposals that support long-term initiatives that need to be repeated every certain period of time to continue funding the work. 

We think Osmotic, a hackathon project to expand the use cases of CV, offers a better alternative for the above proposals.

## Test pilot

**Osmotic hackathon**

[Osmotic Funding · ETHGlobal Showcase](https://showcase.ethglobal.com/ethonline2021/osmotic-funding)

**Source Code**

https://github.com/BlossomLabs/osmotic-funding

## Spec

There are different ways in which we can integrate the changes made on Osmotic Funding to Gardens. Some we thought were:

### Use signaling proposals + superfluid Aragon app

One way we can leverage the concepts of Osmotic without too many changes is by using CV signaling proposals as input of a middleware contract that handles the `fluid proposals` and talks with the recently developed Superfluid Aragon app to `start` and `stop` flows for a beneficiary address once Osmotic threshold is reached.

Note that for this solution we will need a service to keep the middleware contract up to date with the latest relevant signaling proposals data.

**Superfluid Aragon app source code**

[https://github.com/BlossomLabs/superfluid-aragon-app](https://github.com/BlossomLabs/superfluid-aragon-app)

### Update CV

Another alternative we have is to update the existent Conviction Voting app to include relevant methods to create a new proposal type (`fluid`) and the logic for the new calculations.

Some open questions for this alternative are:

- Should we update the existent Gardens CV or install a new app

## Giving more context

The above system is composed of several smart contract technologies. On the core of the whole Gardens we have [AragonOS](https://hack.aragon.org/docs/aragonos-intro.html) the smart contract framework to develop Decentralized Autonomous Organizations. 

Going one level deep we will interact with two existent Aragon applications (smart contracts built on top of AragonOS to manage a specific responsibility on the DAO):
- [Conviction Voting](https://github.com/1Hive/conviction-voting-app) 
- [Superfluid](https://github.com/BlossomLabs/superfluid-aragon-app)

In particular, the Superfluid Aragon app is a wrapper that allow Aragon organizations to use the [Superfluid protocol](https://www.superfluid.finance/home).

**Comment by kamikazebr** [comment]
Awesome ideas and project (Osmotic Funding).

Which pros and cons in use differente new aragon app istead update the current CV?

**Reply by kanaagawa** [reply]
Glad you like it!

About the pros and cons, some that come to mind are:

**Pros**
- Prevent changing current CV that is already in use and battle-tested for more than a year
- Leverage SuperFluid to handle stream payments
- A more modular solution that can be `turn-off` if not have the desired outcome

**Cons**
- Keep track of support amount of signaling proposals linked to fluid proposals
- Develop and maintain a separate service

**Reply by kamikazebr** [reply]
> * Keep track of support amount of signaling proposals linked to fluid proposals
> * Develop and maintain a separate service

In the case both can be solved creating the separate service to it, right?

**Comment by rperez89** [comment]
Hey @0xGabi love this kind of ideas to leverage the gardens protocol. 
A couple of ideas to discuss and how i would suggest to integrate it in order to avoid having kind of a "Frankestain" architecture mixing  apps functionalities (like using the signaling conviction voting to execute something else in another app).

1- Add the superfluid aragon app to the gardens template.
2- Add the necessary entities and handler on the gardens subgraph to keep track of this new app (the same way we integrated the decission voting to the honey pot subgraph).
3- Modify the  frontend to handle this new type of proposals.

This way while it takes a bit more of time i think is a cleaner solution.

Let me know what you think

**Reply by kanaagawa** [reply]
As we discussed async we still need more clarity on the final implementation details about how to integrate Osmotic in the Gardens framework. Even though you are correct in looking for a cleaner solution we also want to reduce as much as possible the smart contract changes in our current apps.

**Comment by Jeff-Emmett** [comment]
Sounds like Osmotic Funding offers a tooling distinction (distinct from CV with threshold) to better handle Operational Expenses (OpEx) like recurring staff costs as opposed to Capital Expenses (CapEx) like one off projects or other discrete investments.

Very cool stuff, fully in support!
